### PR TITLE
CONTRIBUTING.md updates

### PR DESCRIPTION
**Pull request by spectranaut**
for @rwaldron and @leobalter and @jugglinmike !

I tried to update the contributing file a few places to reflect the advice I've gotten from all of you. After these edits I think the file is up-to-date, but it might be worth it for you all to take a quick read through the whole file.

One section I think is lacking is the "features" front matter tag. We should have some concrete recommendations about what necessitates the creation of a new feature (probably, for example, a new feature proposal to ECMA script will necessitate the creation of a new feature tag -- to be used in all the new tests for the new feature).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -49,21 +55,24 @@ Test262 supports the following tags:
...
+
 #### description
 **description**: [string]
 
-This is one of two required frontmatter tags. It should be a short, one-line
-description of the purpose of this testcase.  This is the string displayed by
-the browser runnner.
+This is one of two required frontmatter tags. It should be a short (less than
+100 characters), one-line description of the purpose of this testcase.  This 
+is the string displayed by the browser runnner.
```
Comment (leobalter) [inline CONTRIBUTING.md]: > This is the string displayed by the browser runnner.

We should remove this.

Maybe the 100 characters limit should be a relaxed as a convention, not a rule. What do you think?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -168,36 +165,41 @@ This tag is for boolean properties associated with the test.
 
 Some tests require the use of language features that are not directly described by the test file's location in the directory structure. These features should be specified with this tag. See the `features.txt` file for a complete list of available values.
 
+#### es5id
+**es5id**: [es5-test-id]
+
+**Deprecated.**
+
+This tag identifies the section number from the portion of the ECMAScript 5.1 or ECMAScript 3 standard that is tested by this test. It was automatically generated for tests that were originally written for the ES5 (or earlier) version of the test suite and are now part of the ES6 version. You can use the es5id to discover the relevant portion of the ECMAScript standard by looking up the section number in previous publications of the specification:
```
Comment (leobalter) [inline CONTRIBUTING.md]: remove the `or ECMAScript 3`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -168,36 +165,41 @@ This tag is for boolean properties associated with the test.
...
 assert.sameValue(actual, expected, message) | throw a new Test262Error instance if the first two arguments are not [the same value](https://tc39.github.io/ecma262/#sec-samevalue); accepts an optional string message for use in creating the error
 assert.notSameValue(actual, unexpected, message) | throw a new Test262Error instance if the first two arguments are [the same value](https://tc39.github.io/ecma262/#sec-samevalue); accepts an optional string message for use in creating the error
 assert.throws(expectedErrorConstructor, fn, message) | throw a new Test262Error instance if the provided function does not throw an error, or if the constructor of the value thrown does not match the provided constructor
 assert.throws.early(expectedErrorConstructor, code) | throw a new Test262Error instance if the provided code does not throw an early error, or if the constructor of the value thrown does not match the provided constructor. This assertion catches only errors that will be parsed through `Function(code)`.
+$ERROR(message) | construct a Test262Error object and throw it <br>**DEPRECATED** -- do not use new tests
```
Comment (leobalter) [inline CONTRIBUTING.md]: s/do not use new tests/do not use in new tests

or maybe tell to use one of the assertions or throw a `new Test262Error` instead?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -239,15 +241,15 @@ var p = new Promise(function () { /* some test code */ });
 
 p.then(function checkAssertions(arg) {
   if (!expected_condition) {
-    $ERROR("failure message");
+    throw new Test262Error("failure message");
```
Comment (leobalter) [inline CONTRIBUTING.md]: `assert(expected_condition, "assertion message");` is even better than the whole if case.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -239,15 +241,15 @@ var p = new Promise(function () { /* some test code */ });
...
   }
 
 }).then($DONE, $DONE);
 ```
 
 Function `checkAssertions` implicitly returns `undefined` if the expected condition is observed.  The return value of function `checkAssertions` is then used to asynchronously invoke the first function of the final `then` call, resulting in a call to `$DONE(undefined)`, which signals a passing test.
 
-If the expected condition is not observed, function `checkAssertions` throws a `Test262Error` via function $ERROR.  This is caught by the Promise and then used to asynchronously invoke the second function in the call -- which is also `$DONE` -- resulting in a call to `$DONE(error_object)`, which signals a failing test.
+If the expected condition is not observed, function `checkAssertions` throws a `Test262Error`.  This is caught by the Promise and then used to asynchronously invoke the second function in the call -- which is also `$DONE` -- resulting in a call to `$DONE(error_object)`, which signals a failing test.
```
Comment (leobalter) [inline CONTRIBUTING.md]: this description makes it compelling to leave the `throw new Test262Error` above, we can also consider a failing assertion will throw an exception, being caught by the Promise.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -10,12 +10,18 @@
...
 
 - `Math.fround` handling of `Infinity`: `test/built-ins/Math/fround/Math.fround_Infinity.js`
-- `Array.prototype.find` use with `Proxy`: `test/Array/prototype/find/Array.prototype.find_callable-Proxy-1.js`
+- `Array.prototype.find` use with `Proxy`: `test/built-ins/Array/prototype/find/Array.prototype.find_callable-Proxy-1.js`
 - `arguments` implements an `iterator` interface: `test/language/arguments-object/iterator-interface.js`
 
+See the following directory trees for further recommended examples:
+
+- `test/built-ins/Math/fround`
+- `test/built-ins/weakmap`
```
Comment (leobalter) [inline CONTRIBUTING.md]: `WeakMap` for case sensitive file systems.

File: CONTRIBUTING.md (hunk 1 above)
Comment (rwaldron) [reply CONTRIBUTING.md]: I think we can relax it to a "suggestion"?

File: CONTRIBUTING.md (hunk 2 above)
Comment (rwaldron) [reply CONTRIBUTING.md]: @leobalter it's actually totally accurate.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -10,12 +10,18 @@
...
+- `test/language/arguments-object`
```
Comment (rwaldron) [inline CONTRIBUTING.md]: As we discussed offline, these would make great links

File: CONTRIBUTING.md (hunk 2 above)
Comment (rwaldron) [reply CONTRIBUTING.md]: Might be useful to link to the report https://github.com/tc39/test262/wiki/Test262-Technical-Rationale-Report,-October-2017#specification-reference-ids

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -297,14 +297,13 @@ Test cases and test templates specify meta-data using the same YAML frontmatter
 - test cases (`*.case`)
   - `template` - name of the sub-directory to locate templates for this test
   - `description` (see above)
+    // looks like this is actually "desc" !!! add how its made description with the appending of "name" in parens from the test template
```
Comment (leobalter) [inline CONTRIBUTING.md]: +1

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -297,14 +297,13 @@ Test cases and test templates specify meta-data using the same YAML frontmatter
...
   - `info` (see above)
   - `features` (see above; merged with value defined by test template)
 - test templates (`*.template`)
   - `path` - location within the published test hierarchy to output files created from this template
   - `name` - human-readable name of the syntactic form described by this template (used to generate the test file's `description` field)
   - `esid` (see above)
-  - `es5id` (see above)
-  - `es6id` (see above)
```
Comment (leobalter) [inline CONTRIBUTING.md]: 🎉

**Comment by spectranaut** [comment]
I removed "WIP" from the title because these changes are ready to merge, I think! @leobalter @rwaldron

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -292,19 +295,23 @@ Some language features are expressed through a number of distinct syntactic form
...
+
+### test templates (`*.template`)
+Field | Description
+------|-------------
+`path` | location within the published test hierarchy to output files created from this template. This path will be ended with the name of the test case file. If path is "/test/language/template1-" and the test case is "cast1.js", the final location of the file will be "/test/language/template1-case1.js"
+`name` | human-readable name of the syntactic form described by this template. This text will be appended, in parentheses, to the end of the test cases `desc` field.
+`esid` | see the frontmatter definition of the "info" tag.
+`info` | see the frontmatter definition of the "info" tag. The generated test will have a have final "info" value which is this text concatenated at the end of the test case's "info" text.
+`features` | see the frontmatter definition of the "features" field. The generated test will have a final feature list in combination with the test case's feature field.
+any other valid frontmatter field | see the frontmatter definitions.
```
Comment (rwaldron) [inline CONTRIBUTING.md]: Tables!! I love it.

**Pull request merged**